### PR TITLE
add case for code-quality

### PR DESCRIPTION
**Pull request by wuhuiluo**
## What this PR does / why we need it:
add case for code-quality

## Does this PR introduce a user interface change?

❎ No

## ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |              |
| 🇨🇳 中文    |              |

## Does this PR need be patched to older version?
❎ No

## Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: shell/auto_test/tests/dop/sonar-test.spec.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    await expectExist('text=code defect6D', 0);
+    await expectExist('text=code vulnerability0A', 0);
+    await expectExist('text=code smell22A', 0);
+    await page.click('text=3dacc1');
+    expect(page.url()).toBe(
+      'https://erda.hkci.terminus.io/erda/dop/projects/1/apps/16/repo/commit/3dacc11ac8439f3f63df02be32364731821df0aa',
+    );
+    await wait(1);
+    await page.goto('https://erda.hkci.terminus.io/erda/dop/projects/1/apps/16/test/quality');
+    await page.click('text=coverage 0%');
```
Comment (Zero-Rock) [inline shell/auto_test/tests/dop/sonar-test.spec.ts:18]: don't use selectors with side effects

File: shell/auto_test/tests/dop/sonar-test.spec.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    await page.click('text=market.tsx');
+    await wait(1);
+    await page.click('text=repeat rate 1.5%');
```
Comment (Zero-Rock) [inline shell/auto_test/tests/dop/sonar-test.spec.ts:21]: don't use selectors with side effects

File: shell/auto_test/tests/dop/add-ticket.spec.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    // Click textarea[name="textarea"]
+    await page.click('textarea[name="textarea"]');
+    // Fill textarea[name="textarea"]
+    await page.fill('textarea[name="textarea"]', testData.title);
+    // Click .button.button-type-annex
+    await page.click('.button.button-type-annex');
+    // Click span[role="button"]:has-text("image upload")
+    await page.click('span[role="button"]:has-text("image upload")');
+    // Upload Erda.png
+    await page.setInputFiles('[type="file"]', testData.image);
```
Comment (Zero-Rock) [inline shell/auto_test/tests/dop/add-ticket.spec.ts:30]: use base po

File: shell/auto_test/tests/dop/sonar-test.spec.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+      page.waitForNavigation(/*{ url: 'https://erda.hkci.terminus.io/erda/dop/projects/1/apps/16/ticket/all?pageNo=1&priority=medium&type=bug' }*/),
+      page.click(':nth-match(:text("medium"), 2)'),
+    ]);
+    await page.click(':nth-match(:text("medium"), 2)');
+    await Promise.all([
+      page.waitForNavigation(/*{ url: 'https://erda.hkci.terminus.io/erda/dop/projects/1/apps/16/ticket/all?pageNo=1&priority=high&type=bug' }*/),
+      page.click(':nth-match(:text("high"), 2)'),
+    ]);
+    await page.click('[placeholder="filter by title"]');
+    await page.fill('[placeholder="filter by title"]', 'missing');
```
Comment (Zero-Rock) [inline shell/auto_test/tests/dop/sonar-test.spec.ts:50]: missing assert

File: shell/auto_test/tests/dop/problem-detail.spec.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    // Go to https://erda.hkci.terminus.io/erda/dop/projects/1/apps/16/ticket/open/28
+    await page.goto('https://erda.hkci.terminus.io/erda/dop/projects/1/apps/16/ticket/open/28');
+    await wait(1);
+    // Click text=Issues
+    await Promise.all([
+      page.waitForNavigation(/*{ url: 'https://erda.hkci.terminus.io/integration/dop/projects/123/apps/788/ticket/open?pageNo=1' }*/),
+      page.click('text=Issues'),
+    ]);
+    // Click text=#58
+    await page.click('text=#60');
```
Comment (Zero-Rock) [inline shell/auto_test/tests/dop/problem-detail.spec.ts:20]: don't use selectors with side effects

**Comment by Zero-Rock** [comment]
license

File: shell/app/views/index.ejs [hunk 6]
```diff
@@ -37,8 +37,6 @@
   </script>
   <!-- dice -->
   <script src="//at.alicdn.com/t/font_500774_kzpe8fzg3t.js" async></script>
-  <!-- layout -->
-  <script src="https://lf1-cdn-tos.bytegoofy.com/obj/iconpark/icons_1776_8.ba89731a7541ef588c93a4ace53a6b3d.es5.js" ></script>
```
[the remark this answers is not in the document]
Comment (wuhuiluo) [reply shell/app/views/index.ejs:41]: Yes,Shenghui said this piece could be deleted

**Pull request merged**